ddtrace/internal/telemetry/telemetry_writer.py

```diff
@@ -0,0 +1,239 @@
+from collections import namedtuple
+import logging
+from typing import ClassVar
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ddtrace.internal import forksafe
+
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .telemetry_request import create_telemetry_request
+
+
+TelemetryRequest = namedtuple("TelemetryRequest", "request fail_count")
```

Comment:
Can probably type this with something like


```python
from typing import NamedTupled

class TelemetryRequest(NamedTuple):
    request: Request
    fail_count: int
```

Reply:
I just removed retries from this PR. I'll create a separate jira card for this